wrappers/common/common.go

```diff
@@ -61,15 +61,24 @@ func StartSpanOrTraceFromHTTPWithTraceParserHook(r *http.Request, parserHook con
 		// there is no trace yet. We should make one! and use the root span.
 		var tr *trace.Trace
 		if parserHook == nil {
-			beelineHeader := r.Header.Get(propagation.TracePropagationHTTPHeader)
-			prop, _ := propagation.UnmarshalHoneycombTraceContext(beelineHeader)
+			beelineHeaderValue := r.Header.Get(propagation.TracePropagationHTTPHeader)
+			w3cHeaderValue := r.Header.Get(propagation.TraceparentHeader)
+			var prop *propagation.PropagationContext
+			if beelineHeaderValue != "" {
+				prop, _ = propagation.UnmarshalHoneycombTraceContext(beelineHeaderValue)
+			} else if w3cHeaderValue != "" {
+				headers := map[string]string{
```

Comment:
IDK why the `unmarshal` interface takes a `map` rather than `http.Header` but that's a public interface, so I guess I'll remap the headers here 😞

Reply:
the remapping is in the [docs](https://docs.honeycomb.io/getting-data-in/go/beeline/#interoperability-with-opentelemetry), so this is by design 🤷‍♀️

Reply:
That's weird, I think http.header is a `map[string][]string` so should work 🤷🏻‍♂️